cli/cli/helpers/portal_manager/binary_artifact_getter.go

```
@@ -0,0 +1,166 @@
+package portal_manager
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"github.com/google/go-github/v50/github"
+	"github.com/kurtosis-tech/kurtosis/cli/cli/helpers/host_machine_directories"
+	"github.com/kurtosis-tech/stacktrace"
+	"github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+)
+
+const (
+	kurtosisTechGithubOrg        = "kurtosis-tech"
+	kurtosisPortalGithubRepoName = "kurtosis-portal"
+
+	githubAssetExtension = ".tar.gz"
+
+	portalBinaryFileMode  = 0700
+	portalVersionFileMode = 0600
+)
+
+// DownloadLatestKurtosisPortalBinary downloads the latest version of Kurtosis Portal.
+// It returns true if a new version was downloaded and installed, false if it no-oped because latest was already installed
+// If no error occurs, it also returns the path to the binary file just installed
+func DownloadLatestKurtosisPortalBinary(ctx context.Context) (bool, error) {
+	binaryFilePath, err := host_machine_directories.GetPortalBinaryFilePath()
+	if err != nil {
+		return false, stacktrace.Propagate(err, "Unable to get file path to Kurtosis Portal binary file")
+	}
+	currentVersionFilePath, err := host_machine_directories.GetPortalVersionFilePath()
+	if err != nil {
+		return false, stacktrace.Propagate(err, "Unable to get file path to Kurtosis Portal version file")
+	}
+
+	ghClient := github.NewClient(http.DefaultClient)
+	latestRelease, err := getLatestRelease(ctx, ghClient)
+	if err != nil {
+		return false, stacktrace.Propagate(err, "Unable to retrieve latest release of Kurtosis Portal from Github repository")
+	}
+
+	latestVersionStr, currentVersionStr, currentVersionMatchesLatest, err := compareLatestVersionWithCurrent(currentVersionFilePath, latestRelease)
+	if err != nil {
+		return false, stacktrace.Propagate(err, "Error checking if latest version matches current")
```

similar here; maybe someone is connected to their intranet but have troubles with internet/GitHub is down.

---

Yup, I changed that, we're not failing hard anymore if a version is currently installed 👍